wallet-new/src/Cardano/Wallet/Kernel/CoinSelection/Generic/Fees.hs

```diff
@@ -90,18 +90,26 @@ senderPaysFee pickUtxo totalFee css = do
     let (css', remainingFee) = feeFromChange totalFee css
     (css', ) <$> coverRemainingFee pickUtxo remainingFee
 
-coverRemainingFee :: forall utxo e m. (Monad m, CoinSelDom (Dom utxo))
-                  => (Value (Dom utxo) -> CoinSelT utxo e m (UtxoEntry (Dom utxo)))
+coverRemainingFee :: forall utxo m. (Monad m, CoinSelDom (Dom utxo))
+                  => (Value (Dom utxo) -> CoinSelT utxo CoinSelHardErr m (UtxoEntry (Dom utxo)))
                   -> Fee (Dom utxo)
-                  -> CoinSelT utxo e m (SelectedUtxo (Dom utxo))
+                  -> CoinSelT utxo CoinSelHardErr m (SelectedUtxo (Dom utxo))
```

Comment:
This is the only thing I don't like about this PR: Edsko worked really hard to make the code completely generic, even in the error `e`, but here we are now specialising it to `CoinSelHardErr`. Is there any way we can keep it generic? It would be better, as I also don't have any clue if this would preclude along the lines the possibility to run more interesting simulations, for example.

Reply:
Right, good point. `coverRemainingFee` is used by `senderPaysFee` which is _already_ specialised, so I guess this is fine. Alright 😉